src/Microsoft.Identity.Abstractions/Results/OperationResult.cs

```diff
@@ -19,8 +21,7 @@ namespace Microsoft.Identity.Abstractions
     /// This pattern helps with clear error handling and propagation.
     /// </remarks>
     public readonly record struct OperationResult<TResult, TError>
-            where TResult : class
-            where TError : OperationError
+        where TError : OperationError

```

Comment:
Can you add more tests to https://github.com/AzureAD/microsoft-identity-abstractions-for-dotnet/blob/main/test/Microsoft.Identity.Abstractions.Tests/OperationResultTests.cs that use primitive types?